src/server/stream_family.cc

```diff
@@ -817,6 +817,26 @@ OpStatus OpDestroyGroup(const OpArgs& op_args, string_view key, string_view gnam
   return OpStatus::SKIPPED;
 }
 
+// XGROUP CREATECONSUMER key groupname consumername
+OpResult<uint32_t> OpCreateConsumer(const OpArgs& op_args, string_view key, string_view gname,
+                                    string_view consumer_name) {
+  OpResult<pair<stream*, streamCG*>> cgroup_res = FindGroup(op_args, key, gname);
+  if (!cgroup_res)
+    return cgroup_res.status();
+  streamCG* cg = cgroup_res->second;
+  if (cg == nullptr)
+    return OpStatus::SKIPPED;
+
+  auto* shard = op_args.shard;
+  shard->tmp_str1 = sdscpylen(shard->tmp_str1, consumer_name.data(), consumer_name.size());
+  streamConsumer* consumer =
+      streamCreateConsumer(cg, shard->tmp_str1, NULL, 0, SCC_NO_NOTIFY | SCC_NO_DIRTIFY);
+
+  if (consumer)
+    return OpStatus::OK;
+  return OpStatus::KEY_EXISTS;
```

Comment:
Seems like we use SKIPPED for when cg is null, that is in what case? If the key is not found? Or is that part of `if (!cgroup_res)`?

Anyways, just a minor nit: using skipped here would maybe be more accurate

Reply:
It is to check if the group with the given groupname exists. I also wanted to use OpStatus::SKIPPED, but using the same status for two different reason is not possible here.

The redis doc says if the consumer already exists, it will send '0' as output. That's why it is required to SendLong(0) if the consumer already exists.